bat-utils/lib/hapi-server.js

```diff
@@ -66,194 +74,112 @@ const Server = async (options, runtime) => {
-                return { limit: Number.MAX_SAFE_INTEGER, window: 1 }
-              }
-
-              if (whitelist.authorizedP(ipaddr)) {
-                authorization = request.raw.req.headers.authorization
-                if (authorization) {
-                  parts = authorization.split(/\s+/)
-                  token = (parts[0].toLowerCase() === 'bearer') && parts[1]
-                } else {
-                  token = request.query.access_token
-                }
-                tokenlist = process.env.TOKEN_LIST ? process.env.TOKEN_LIST.split(',') : []
-                limit = (typeof token === 'string' && braveHapi.isSimpleTokenValid(tokenlist, token)) ? 60000 : 3000
-              }
-            } catch (e) {
-            }
+        plugin: hapiRequireHTTPS,
+        options: { proxy: true }
+      }
+    ] : []
+  )
+  await server.register(plugins)
+
+  debug('extensions registered')
+
+  if (runtime.login) {
+    if (runtime.login.github) {
+      const { github } = runtime.login
+      server.auth.strategy('github', 'bell', {
+        provider: 'github',
+        password: cryptiles.randomString(64),
+        clientId: github.clientId,
+        clientSecret: github.clientSecret,
+        isSecure: github.isSecure,
+        forceHttps: github.isSecure,
+        scope: ['user:email', 'read:org'],
+        location: (process.env.HOST.startsWith('127.0.0.1') ? 'http://' : 'https://') + process.env.HOST
+      })
 
-            return { limit: limit, window: 60 }
-          },
-          enabled: true,
-          methods: [ 'get', 'post', 'delete', 'put', 'patch' ],
-          overLimitError: (rate) => boom.tooManyRequests(`try again in ${rate.window} seconds`),
-          rateLimitKey: (request) => {
-            try {
-              return whitelist.ipaddr(request) + ':' + runtime.config.server.host
-            } catch (e) {
-              return 'default'
-            }
-          },
-          redisClient: (runtime.cache && runtime.cache.cache) || runtime.queue.config.client
-        }
-      },
-      {
-        register: swagger,
-        options: {
-          auth: {
-            strategy: 'whitelist',
-            mode: 'required'
-          },
-          info: {
-            title: npminfo.name,
-            version: npminfo.version,
-            description: npminfo.description
-          }
+      debug('github authentication: forceHttps=' + github.isSecure)
+
+      server.auth.strategy('session', 'cookie', {
+        cookie: {
+          password: github.ironKey,
+          isSecure: github.isSecure
         }
-      }
-    ], async (err) => {
-      if (err) {
-        debug('unable to register extensions', err)
-        reject(err)
-        throw err
-      }
+      })
 
+      debug('session authentication strategy via cookie')
+    } else {
+      debug('github authentication disabled')
       if (process.env.NODE_ENV === 'production') {
```

Comment:
IMO all checks for `production` should check not `development` since I believe `staging` to be `the next production` and we need to make sure it will behave as in `production`

Reply:
Resolved by @evq comment above